src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1230,12 +1290,14 @@
       (merge (select-keys certificate-authority (keys CaSettings)))
       (initialize-ca-config)
       (assoc :ruby-load-path (:ruby-load-path jruby-puppet)
+             :auto-renewal-cert-ttl (duration-str->sec (:auto-renewal-cert-ttl certificate-authority))
```

Comment:
We need `allow-auto-renewal` to be extracted from the config and added to the structure too.

Reply:
allow-auto-renewal will be added in the initialize-ca-config here https://github.com/puppetlabs/puppetserver/pull/2728/files#diff-ac2e29a707d8bb0a3cca1bdb78e91e56a4a9a9213fb1fde7839b628837f0cab1R259 right?

Reply:
Yes, the default is set there.  This is the point where we override the defaults with the values from the configuration file.